AddonTerraformTemplate/main.tf

```diff
@@ -90,6 +90,18 @@ resource "azurerm_monitor_data_collection_rule_association" "dcra" {
   ]
 }
 
+resource "azurerm_monitor_data_collection_rule_association" "dcraprivatelink" {
+  count = var.useAzureMonitorPrivateLinkScope && azurerm_kubernetes_cluster.k8s.location == azurerm_monitor_workspace.amw.location ? 1 : 0
```

Comment:
What about the scenario where the region does not match? Do you want to request the customer for a 
DCE id as a parameter maybe and populate this field accordingly?

Reply:
Thanks Kaveesh! Yes that makes sense. let me update the template.

Reply:
Updated in the latest commit.